versioned_docs/version-latest/protocol/Partition-mount.md

```diff
@@ -0,0 +1,411 @@
+
+## VFAT
+#### Creating FAT32 partition. It is recommended for boot in GPT:
+```console
+mkfs.vfat -c -n "boot" -F 32 -v /dev/nvme1n1p1  # c -check for errors; n-label; F -FAT size, v-verbose
+```
+
+## EXT(3,4)
+#### creating EXT4 filesystem with setting it's name (farmer-0):
+```console
+mkfs.ext4 -L "farmer-0" /dev/nvme0n1p1  # -L - label
+```
+#### Creating small EXT4 partition which will be used as external journal:
+###### note: external-journal have same block size as it's main filesystem (normally, it is so by default)
+```console
+mke2fs -L "j0" -O journal_dev /dev/nvme1n1p3
+```
+#### attaching the filesystem to the journal device located on /dev/nvme1n1p3.
+```console
+tune2fs -J device=/dev/nvme1n1p3 /dev/nvme0n1p1
+```
+[detailed EXT4 manual](https://linux.die.net/man/5/ext4)
+
+#### Resizing journal partition: 
+```console
+tune2fs -J size=256M /dev/device
+```
+Some useful options to add in command:
+ -t filesystem type
+ -c	checking for errors (-cc -- very thorough checking)
+ -b	set block size 4096
+ -v	verbose execution 
+ 
+To check/veryfy/repair external journal partition :
+```console
+e2fsck -j /dev/nvme1n1p3 /dev/nvme1n1p2  # Requires to unmount partition
+```
+
+It is allowed to switch journal on EXT4 partition from external to internal. 
+```console
+umount /dev/sdXY	    # unmount
+tune2fs -j /dev/sdXY	# create inner journal
+e2fsck -f /dev/sdXY	    # check file system
+mount /dev/sdXY /mount/point  
+```
+
+[detailed e2fsck manual](https://linux.die.net/man/8/e2fsck)
+
+
+
+## XFS
+* XFS provides various file system parameters that can be adjusted using the xfs_admin 
+* You can modify the mount options for an XFS file system to change its behavior.
+* XFS provides additional performance tuning options: delaylog ; allocsize ; inode64, You modify these options by editing the `/etc/default/grub`
+* When creating XFS file system, you should specify an external log device using the `-l logdev=device` option with the `mkfs.xfs` command. 
+If you do not specify a log device duiring creating the file system, you will not be able to add one later
+* You don't need to mount the journal separately! - Just the main partition mount is enough. The log will be hooked up automatically.
+
+Some interesting functions of XFS filsystem:
+    
+    -l log_section_options (These options specify the location, size, and other parameters of the log section of the filesystem. The valid log_section_options are: `logdev=device`. This is used to specify that the log section should reside on the device separate from the data section.
+
+    nobarrier (For ex. : `mount -o nobarrier /dev/device /mount/point`) For devices with non-volatile, battery-backed write caches and those with write-caching disabled, you can safely disable write barriers at mount time using the -o nobarrier option for mount. The role of write barriers in maintaining data integrity during power failures when dealing with storage devices that have extra caches.
```

Comment:
Wrong formatting, this is shown as a code block, while a lot of it is a plaint text. This is the case for the next 3 sections.